src/Simulation/Simulators/ToffoliSimulator/ToffoliSimulator.cs

```diff
@@ -483,4 +493,67 @@ public override void Apply(IQArray<Qubit> qubits)
             }
         }
     }
+    
+    internal static class Extensions
```

Comment:
do we want to do any null checks?

Reply:
Good point, yeah. I can use `#nullable enable` as well to help make sure I've caught relevant null-checks.